### PR TITLE
Create direct message respect the user role (Guest/Normal user)

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Create direct message respect the user role (Guest/Normal user)

#### Ticket Link
[MM-17195](https://mattermost.atlassian.net/browse/MM-17195)

### CODE REVIEW

File: app/channel.go [hunk 1]
```diff
@@ -340,15 +340,19 @@ func (a *App) createDirectChannel(userId string, otherUserId string) (*model.Cha
 		close(uc2)
 	}()
 
-	if result := <-uc1; result.Err != nil {
+	result := <-uc1
+	if result.Err != nil {
 		return nil, model.NewAppError("CreateDirectChannel", "api.channel.create_direct_channel.invalid_user.app_error", nil, userId, http.StatusBadRequest)
```
Comment (mkraft) [inline app/channel.go]: Did you intend to leave the details parameter as `userId` or were you intending to add `result.Err.Error()`?

File: app/channel.go [hunk 2]
```diff
@@ -340,15 +340,19 @@ func (a *App) createDirectChannel(userId string, otherUserId string) (*model.Cha
...
 	}
+	user := result.Data.(*model.User)
 
-	if result := <-uc2; result.Err != nil {
+	result = <-uc2
+	if result.Err != nil {
 		return nil, model.NewAppError("CreateDirectChannel", "api.channel.create_direct_channel.invalid_user.app_error", nil, otherUserId, http.StatusBadRequest)
```
Comment (mkraft) [inline app/channel.go]: Same question here.

File: app/channel.go (hunk 1 above)
Comment (jespino) [reply app/channel.go]: I wasn't changing anything there, so I was just leaving it as was before.

Comment (mkraft) [reply app/channel.go]: Ok, just seems weird but not related to this PR.

**Pull request merged**